modules/services/ui/src/main/java/au/com/shiftyjelly/pocketcasts/ui/helper/AppIcon.kt

```diff
@@ -218,19 +218,30 @@ class AppIcon @Inject constructor(
     val allAppIconTypes = AppIconType.values()
 
     fun enableSelectedAlias(selectedIconType: AppIconType) {
-        val componentPackage =
-            if (BuildConfig.DEBUG) "au.com.shiftyjelly.pocketcasts.debug" else "au.com.shiftyjelly.pocketcasts"
+        // Ensure the new icon is enabled before the old one in removed to
+        // prevent entering a state where there's no launcher activity
+        setIconEnabledFlag(selectedIconType, true)
+
+        AppIconType.values()
+            .filter { icon -> icon != selectedIconType }
+            // On debug builds always keep the default activity so it can be launched from Android Studio
```

Comment:
I had this run on release variants only. Otherwise if you change the app icon, Android Studio will continue to launch the default `MainActivity` intent even though it's been changed.

You can fix this by either changing the Launch Options or keeping both icons for debug builds. Figured this approach is easier during development.
<img width="1152" alt="Screenshot 2023-09-30 at 10 00 19 PM" src="https://github.com/Automattic/pocket-casts-android/assets/6628497/dfb06bfd-993f-47c2-9769-4397c42d2c9b">